internal/provider/resources/service_account.go

```diff
@@ -244,10 +242,7 @@ func (r *ServiceAccountResource) Read(ctx context.Context, req resource.ReadRequ
 		return
 	}
 
-	resp.Diagnostics.Append(copyServiceAccountToModel(ctx, serviceAccount, &model)...)
```

Comment:
we dont return a diagnostic in the helper, so we dont need to pass it into `Append`